Automattic/woocommerce-payments
# Show uncaptured transactions tab only when some criteria is met

Fixes #5051 

#### Changes proposed in this Pull Request

1. Auth & capture setting is turned on (this is a global setting)
2. Merchant turned on "authorize and capture later" setting 
OR
2. There are some authorizations that need to be captured (even if the merchant turned off the "authorize and capture later" setting)

#### Testing instructions

There are 4 scenarios to test (also captured by test cases)

Default scenario:
- Get the PR
- The Uncaptured section will not show by default (the global setting is turned off)
- Go to `includes/class-wc-payments-features.php`
- Change this line in preparation for next scenarios
```php
	public static function is_auth_and_capture_enabled() {
		return '1' === get_option( self::AUTH_AND_CAPTURE_FLAG_NAME, '0' );
	}
// to
	public static function is_auth_and_capture_enabled() {
		return '1' === get_option( self::AUTH_AND_CAPTURE_FLAG_NAME, '1' );
	}
```

Scenario 1:
- Manual capture **ON** (can be done from Payments settings)
- **Some authorizations** to capture
- Uncaptured tab **should show**

Scenario 2:
- Manual capture **ON** (can be done from Payments settings)
- **No authorizations** to capture
- Uncaptured tab should **show**

Scenario 3:
- Manual capture **OFF** (can be done from Payments settings)
- **Some authorizations** to capture
- Uncaptured tab should **show**

Scenario 4:
- Manual capture **OFF** (can be done from Payments settings)
- **No authorizations** to capture
- Uncaptured tab should **not show**

-------------------


**Post merge**
